src/main/java/application/SessionController.java

```diff
@@ -88,4 +89,40 @@ public CurrentResponseBean getCurrent(@RequestBody String body) throws IOExcepti
         FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
         return new CurrentResponseBean(formEntrySession);
     }
+
+    @RequestMapping(value = "/submit", method = RequestMethod.POST)
+    @ResponseBody
+    public SubmitResponseBean submitForm(@RequestBody String body) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        SubmitRequestBean submitRequestBean = mapper.readValue(body, SubmitRequestBean.class);
+        SerializableSession serializableSession = sessionRepo.find(submitRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+        return new SubmitResponseBean(formEntrySession);
+    }
+
+    @RequestMapping(value = "/get_instance", method = RequestMethod.GET)
+    @ResponseBody
+    public GetInstanceResponseBean getInstance(@RequestBody String body) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        GetInstanceRequestBean getInstanceRequestBean = mapper.readValue(body, GetInstanceRequestBean.class);
+        SerializableSession serializableSession = sessionRepo.find(getInstanceRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+        return new GetInstanceResponseBean(formEntrySession);
+    }
+
+    @RequestMapping(value = "/evaluate_xpath", method = RequestMethod.GET)
+    @ResponseBody
+    public EvaluateXPathResponseBean evaluateXpath(@RequestBody String body) {
+        try {
+            System.out.println("Evaluation called");
+            ObjectMapper mapper = new ObjectMapper();
+            EvaluateXPathRequestBean evaluateXPathRequestBean = mapper.readValue(body, EvaluateXPathRequestBean.class);
+            SerializableSession serializableSession = sessionRepo.find(evaluateXPathRequestBean.getSessionId());
+            FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+            return new EvaluateXPathResponseBean(formEntrySession, evaluateXPathRequestBean.getXpath());
+        } catch(Exception e){
```

Comment:
also would be good to wrap the try catch around the thing we think will fail and catch that specific error if possible